pkg/api/types.go

```diff
@@ -822,6 +822,10 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// PullSecretNames is an optional list of secret names to use for pulling any of the images used by this PodSpec.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	PullSecretNames []string `json:"pullSecretNames,omitempty" description:"list of secret names available for pulling the container images"`
```

Comment:
> I don't think a list of strings works well with patching

`ValidatePodUpdate` would not currently allow changing this... I think maybe it should, since it relates to container images, which are mutable.

> Just choosing a patchStrategy to use as a default or creating a container object with a name?

patchStrategy on a scalar list just merges, which means you can't patch to remove an item. That's why the `Secrets` field for a `ServiceAccount` is currently `[]api.ObjectReference` scrubbed of everything except `Name`